File: backend/src/main/java/be/technobel/parsemaster/validation/constraint/PositiveOrZero.java
package be.technobel.parsemaster.validation.constraint;

import be.technobel.parsemaster.validation.validator.PositiveOrZeroValidator;
import jakarta.validation.Constraint;
import jakarta.validation.Payload;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Target({ElementType.PARAMETER, ElementType.FIELD})
@Retention(RetentionPolicy.RUNTIME)
@Constraint(validatedBy = PositiveOrZeroValidator.class)
public @interface PositiveOrZero {
  String field() default "field";
  String message() default "<0";
  Class<?>[] groups() default {};
  Class<? extends Payload>[] payload() default {};
}

File: src/main/kotlin/be/technobel/playzone/pl/validation/constraints/PositiveOrZero.kt
package be.technobel.playzone.pl.validation.constraints

import be.technobel.playzone.pl.validation.validators.PositiveOrZeroValidator
import jakarta.validation.Constraint
import jakarta.validation.Payload
import kotlin.reflect.KClass

@Target(AnnotationTarget.VALUE_PARAMETER)
@Retention(AnnotationRetention.RUNTIME)
@Constraint(validatedBy = [PositiveOrZeroValidator::class])
annotation class PositiveOrZero(
	val field: String = "field",
	val message: String = "positive_or_zero",
	val groups: Array<KClass<*>> = [],
	val payload: Array<KClass<out Payload>> = []
)